src/ML-POD/pair_pod.cpp

```diff
@@ -0,0 +1,370 @@
+  int *ilist = list->ilist;
+  int nlocal = atom->nlocal;
+  int inum = list->inum;
+  int nall = inum + atom->nghost;
+
+  // initialize global descriptors to zero
+
+  int nd1234 = podptr->pod.nd1234;
+  podptr->podArraySetValue(gd, 0.0, nd1234);
+
+  for (int ii = 0; ii < inum; ii++) {
+    int i = ilist[ii];
+    int jnum = numneigh[i];
+
+    // allocate temporary memory
+
+    if (nijmax < jnum) {
+      nijmax = PODMAX(nijmax, jnum);
+      nablockmax = 1;
+      free_tempmemory();
+      estimate_tempmemory();
+      allocate_tempmemory();
+    }
+
+    // get neighbor pairs for atom i
+
+    lammpsNeighPairs(x, firstneigh, type, map, numneigh, i);
+
+    // compute global POD descriptors for atom i
+
+    podptr->linear_descriptors_ij(gd, tmpmem, rij, &tmpmem[nd1234], numneighsum,
+      typeai, idxi, ti, tj, 1, nij);
+  }
+
+  int nd1 = podptr->pod.nd1;
+  int nd2 = podptr->pod.nd2;
+  int nd3 = podptr->pod.nd3;
+  int nd4 = podptr->pod.nd4;
+  int nd22 = podptr->pod.nd22;
+  int nd23 = podptr->pod.nd23;
+  int nd24 = podptr->pod.nd24;
+  int nd33 = podptr->pod.nd33;
+  int nd34 = podptr->pod.nd34;
+  int nd44 = podptr->pod.nd44;
+  int nd = podptr->pod.nd;
+  bigint natom = atom->natoms;
+
+  for (int j=nd1234; j<(nd1234+nd22+nd23+nd24+nd33+nd34+nd44); j++)
+    newpodcoeff[j] = podcoeff[j]/(natom);
+
+  for (int j=(nd1234+nd22+nd23+nd24+nd33+nd34+nd44); j<nd; j++)
+    newpodcoeff[j] = podcoeff[j]/(natom*natom);
+
+  // compute energy and effective coefficients
+
+  eng_vdwl = podptr->calculate_energy(energycoeff, forcecoeff, gd, newpodcoeff);
+
+  for (int ii = 0; ii < inum; ii++) {
+    int i = ilist[ii];
+
+    // get neighbor pairs for atom i
+
+    lammpsNeighPairs(x, firstneigh, type, map, numneigh, i);
+
+    // compute atomic force for atom i
+
+    podptr->calculate_force(f, forcecoeff, rij, tmpmem, numneighsum,
+      typeai, idxi, ai, aj, ti, tj, 1, nij);
+  }
+
+  if (vflag_fdotr) virial_fdotr_compute();
+
+}
+
+/* ----------------------------------------------------------------------
+   global settings
+------------------------------------------------------------------------- */
+
+void CPairPOD::settings(int narg, char ** /* arg */)
+{
+  if (narg > 0)
+  error->all(FLERR,"Illegal pair_style command");
+}
+
+/* ----------------------------------------------------------------------
+   set coeffs for one or more type pairs
+------------------------------------------------------------------------- */
+
+void CPairPOD::coeff(int narg, char **arg)
+{
+  int n = atom->ntypes;
+  memory->create(setflag,n+1,n+1,"pair:setflag");
+  memory->create(cutsq,n+1,n+1,"pair:cutsq");
+  memory->create(scale,n+1,n+1,"pair:scale");
+  map = new int[n+1];
+  for (int ii = 0; ii < n+1; ii++)
+    for (int jj = 0; jj < n+1; jj++)
+      scale[ii][jj] = 1.0;
+  allocated = 1;
+
+  if (narg != 4 + atom->ntypes) error->all(FLERR,"Incorrect args for pair coefficients");
+
+  map_element2type(narg-4,arg+4);
+
+  //std::cout<<map[0]<<std::endl;
```

Comment:
Please remove commented out debug code unless there is a permanent legitimate need for such blocks (this doesn't look like it). Then either put them into a local/static convenience function, or use #if 0 / #endif.  Or make the #if depend on some local define at the top (say:  #define POD_DEBUG_MAP 0) where you can change the value to trigger getting debug info. You also want to avoid multiple outputs for parallel runs, so check for comm->me == 0.

Reply:
Removed.